File: 0669-trim-a-binary-search-tree/0669-trim-a-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }0
 * }
 */
class Solution {
    public TreeNode trimBST(TreeNode root, int low, int high) {
        //base case 
        if(root == null) return null;
        //Case 1 
        if(root.val < low) return trimBST(root.right,low,high);
        //Case 2
        if(root.val > high) return trimBST(root.left,low,high);
        //Case 3 in range    
        
        root.left = trimBST(root.left,low,high);
        root.right = trimBST(root.right,low,high);
        
        return root;
    }
}

File: 0669-trim-a-binary-search-tree/0669-trim-a-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* trimBST(TreeNode* root, int low, int high) {
        //Base Case
        if(!root) return nullptr;
        //Case 1
        if(root->val < low) return trimBST(root->right,low,high);
        //Case 2
        if(root->val > high) return trimBST(root->left,low,high);
        //Case 3 :- in range
        root->left = trimBST(root->left,low,high);
        root->right = trimBST(root->right,low,high);
        
        return root;
    }
};